src/include/storage/projected_row.h

```diff
@@ -266,6 +270,7 @@ class ProjectedRowInitializer {
  private:
   friend class catalog::Catalog;                  // access to the PRI default constructor
   friend class catalog::DatabaseCatalog;          // access to the PRI default constructor
+  friend class catalog::postgres::PgProcImpl;     // access to the PRI default constructor
```

Comment:
I'm inclined to punt on this. We don't that _that_ many catalog implementation classes, and the documented purpose of having the default constructor for the PRI is to avoid an alleged performance hit. I'll open an issue.

Reply:
#1357